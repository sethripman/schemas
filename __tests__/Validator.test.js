const Validator = require('../lib/Validator');

describe('Validator', () => {
  let validator;

  describe('requiring fields', () => {
    beforeAll(() => {
      validator = new Validator('age', {
        type: Number,
        required: true
      });
    });
  
    it('returns the field', () => {
      const wine = {
        name: 'vinto',
        age: 20,
        price: '1000 buckaroos'
      };
  
      expect(validator.validate(wine)).toEqual(20);
    });

    it('casts field to the inputted type', () => {
      const wine = {
        name: 'vinto',
        age: '20',
        price: '1000 buckaroos'
      };
  
      expect(validator.validate(wine)).toEqual(20);
    });
  
    it('returns absent fields', () => {
      const wine = {
        name: 'vinto',
        price: '1000 buckaroos'
      };
  
      expect(() => validator.validate(wine)).toThrowErrorMatchingSnapshot();
    });
  });
  
  describe('optional fields', () => {
    beforeAll(() => {
      validator = new Validator('age', {
        type: Number
      });
    });
  
    it('returns the field', () => {
      const wine = {
        name: 'vinto',
        age: 20,
        price: '1000 buckaroos'
      };
  
      expect(validator.validate(wine)).toEqual(20);
    });
  
    it('casts field to the inputted type', () => {
      const wine = {
        name: 'vinto',
        age: '20',
        price: '1000 buckaroos'
      };
  
      expect(validator.validate(wine)).toEqual(20);
    });
  
    it('returns null if field is not required', () => {
      const wine = {
        name: 'vinto',
        age: 20,
        price: '1000 buckaroos'
      };
  
      expect(validator.validate(wine)).toBeNull();
    });
  });
});
