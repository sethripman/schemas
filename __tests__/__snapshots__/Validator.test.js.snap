// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator required fields returns the field 1`] = `"age is required"`;

exports[`Validator requiring fields returns absent fields 1`] = `"age is required"`;
